agent/client/client.go

```diff
@@ -0,0 +1,96 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/kubeshop/tracetest/agent/proto"
+	"google.golang.org/grpc"
+)
+
+type Config struct {
+	APIKey    string
+	AgentName string
+}
+
+type SessionConfig struct {
+	BatchTimeout time.Duration
```

Comment:
When connecting to the server, we gonna receive information about how the agent must behave. One of the behaviors that will be configured by the server is how long the agent must batch spans before sending the entire batch when it is acting as a collector.